ddl/ddl_worker.go

```diff
@@ -357,16 +373,41 @@ func setJobStateToQueueing(job *model.Job) {
 	job.State = model.JobStateQueueing
 }
 
-// getHistoryDDLJob gets a DDL job with job's ID from history queue.
-func (d *ddl) getHistoryDDLJob(id int64) (*model.Job, error) {
-	se, err := d.sessPool.get()
-	if err != nil {
-		return nil, errors.Trace(err)
+// addBatchDDLJobs2Table gets global job IDs and puts the DDL jobs in the DDL job table.
+func (d *ddl) addBatchDDLJobs2Table(tasks []*limitJobTask) error {
+	var ids []int64
+	var err error
+	startTs := uint64(0)
+	ctx := kv.WithInternalSourceType(context.Background(), kv.InternalTxnDDL)
+	err = kv.RunInNewTxn(ctx, d.store, true, func(ctx context.Context, txn kv.Transaction) error {
+		t := meta.NewMeta(txn)
+		ids, err = t.GenGlobalIDs(len(tasks))
+		if err != nil {
+			return errors.Trace(err)
+		}
+		startTs = txn.StartTS()
+		return nil
+	})
+	if err == nil {
+		jobTasks := make([]*model.Job, len(tasks))
+		for i, task := range tasks {
+			job := task.job
+			job.Version = currentVersion
+			job.StartTS = startTs
+			job.ID = ids[i]
+			setJobStateToQueueing(job)
+			jobTasks[i] = job
+			injectModifyJobArgFailPoint(job)
+		}
+		sess, err1 := d.sessPool.get()
+		if err1 == nil {
+			sess.SetDiskFullOpt(kvrpcpb.DiskFullOpt_AllowedOnAlmostFull)
```

Comment:
If sessions in ddl all do this, we can put it to `sessPool.get()`?

Reply:
seems not all of them in sessPool need this. better keep the right to the caller

Reply:
When do we need to do this? I see a lot of places do this.

Reply:
most of the is operating ddl job like `getJob`, `updateJob`, but there are many other usages like `gc` and `loadVars` it is not set this